OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -0,0 +1,121 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Automatically transports harvesters with the Carryable trait between resource fields and refineries.")]
+	public class AutoCarryallInfo : CarryallInfo, ITraitInfo, Requires<BodyOrientationInfo>
```

Comment:
`Requires<AircraftInfo>` as well otherwise it will crash in the `PickupUnit` constructor.